api/src/main/java/org/openmrs/module/patientqueueing/api/PatientQueueingService.java

```diff
@@ -26,6 +28,29 @@
  */
 public interface PatientQueueingService extends OpenmrsService {
 	
+	/**
+	 * This method gets a patientQueue by the visitNumber
+	 * 
+	 * @param visitNumber The visitNumber String that is used to search for a patientQueue
+	 * @return a patientQueue that matches the visitNumber passed
+	 * @throws APIException
+	 */
+	@Transactional
+	public PatientQueue getPatientQueueByVisitNumber(String visitNumber);
+	
+	/**
+	 * This Generates a visitNumber based on location and date. The Number generated is unique for a
+	 * patient on a given day A patient will only have one patientvisitNumber for a given day. The
+	 * same number will be reassigned to another queue, incase its the same patient on the same day.
+	 * 
+	 * @param location Location which will be used to first three characters of the visitNumber
```

Comment:
I do not think that the parameter description should talk about the implementation details like first n characters, etc